src/logic/UniqueValues.java

```diff
@@ -0,0 +1,53 @@
+package logic;
+
+import java.util.*;
+
+public class UniqueValues {
+
+    Map<String, Integer> UniqueValuesAndCounters;
+
+    public UniqueValues(ArrayList<String> values) {
+        UniqueValuesAndCounters = new TreeMap<String, Integer>();
+        fillMap(values);
+    }
+
+    private void fillMap(ArrayList<String> values) {
+        for (String val : values) {
+            addUniqueValeuOrIncreaseCounter(val);
+        }
+    }
+
+    private void addUniqueValeuOrIncreaseCounter(String value) {
+        if (UniqueValuesAndCounters.containsKey(value)) {
+            int counter = UniqueValuesAndCounters.get(value);
+            UniqueValuesAndCounters.put(value, ++counter);
+        } else {
+            UniqueValuesAndCounters.put(value, 1);
+        }
+    }
+
+    public void printUniqueValuesAndCounters(boolean onlyNumbers) {
+        Set<Map.Entry<String, Integer>> entries = UniqueValuesAndCounters.entrySet();
+        for (Map.Entry entry : entries) {
+            if (onlyNumbers) {
+                if (isNumeric(entry.getKey().toString())) { // drukuj tylko liczby
+                    System.out.println("'" + entry.getKey() + "'\t\t– liczba wystąpień:\t" + entry.getValue());
+                }
+            } else {
+                System.out.println("'" + entry.getKey() + "'\t\t– liczba wystąpień:\t" + entry.getValue());
+            }
+        }
+    }
+
+    private boolean isNumeric(String strNum) {
+        if (strNum == null) {
+            return false;
+        }
+        try {
+            double d = Double.parseDouble(strNum);
+        } catch (NumberFormatException nfe) {
```

Comment:
korzystanie  z try catcha jako sposób żeby zwrocić rożne wartości to nie jest dobry pomysl jesli spodziewasz sie ze ten wyjatke bedzie rzucany często.  A wygląda n ato ze wyjątek będzie rzucany bradzo często(poniewaz plik wejsciowy sklada sie cyfr i znakow) i to bedzie mialo zły wpływ na wydajność

lepiej uzyc np.  metody z bilbiotek apache commons lang( http://commons.apache.org/proper/commons-lang/apidocs/org/apache/commons/lang3/StringUtils.html#isNumeric-java.lang.CharSequence- )  tak jak omawialismy na zajeciach